cmake/FindCoroutines.cmake

```diff
@@ -115,6 +115,9 @@ check_cxx_compiler_flag(-fcoroutines _CXX_COROUTINES_SUPPORTS_CORO_FLAG)
 
 if(_CXX_COROUTINES_SUPPORTS_MS_FLAG)
     set(_CXX_COROUTINES_EXTRA_FLAGS "/await")
+    if(CMAKE_CL_64)
```

Comment:
Is there some documentation which versions of MSVC support the "/await:heapelide" flag? Are there old versions of MSVC which we could break by unconditionally setting the heapelide flag?
And do the new versions (16.8), which do not need the /await flag, need the heapelide flag?
I'd appreciate if we made sure we do not break anything by setting this flag.

Reply:
I cannot find any documentation about this hidden version of the /await flag, all I know is that MSVC 2019 does not complain about it.

Reply:
Can we maybe, instead of checking for CMAKE_CL_64 (which is discouraged, according to CMake documentation), check for the existence of this flag using check_cxx_compiler_flag? That way we'd make sure nothing bad happens if some version of MSVC does not know it.

Reply:
I didn't know CMAKE_CL_64  was discouraged...
Yes it look like the better way to go

Reply:
Hum, however the /await:heapelide is also ok for 32bit target.
Do you know why have been added for 64bit targets only ?

Reply:
when using the heapelide version, `auto reset event test` also runs into SEGFAULT.

Reply:
no, sorry it's ok, I was targetting wrong processor